### PR TITLE
More flexible services for RPC

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Licini** [comment]
nice!

**Comment by tomvanmele** [comment]
```python
# special.py

from compas.rpc import Server
from compas.rpc import Dispatcher


class SpecialService(Dispatcher):
    def special(self):
        return "special"


def start_service():
    server = Server(("0.0.0.0", 1753))
    service = SpecialService()
    server.register_instance(service)
    server.serve_forever()


if __name__ == "__main__":
    import argparse
    parser = argparse.ArgumentParser()
    args, unknown = parser.parse_known_args()
    start_service()

```

```python
# client.py

import os
from compas.rpc import Proxy

proxy = Proxy(service="special", cwd=os.path.dirname("__file__"))

proxy.special()

```

File: src/compas/rpc/services/default.py [hunk 1]
```diff
@@ -4,67 +4,32 @@
...
-    server = Server(("0.0.0.0", port))
+    host = "0.0.0.0"
+    address = host, port
+    server = Server(address)
 
     # register a few utility functions
-    server.register_function(server.ping)
-    server.register_function(server.remote_shutdown)
+    # server.register_function(server.ping)
+    # server.register_function(server.remote_shutdown)
```
Comment (gonzalocasas) [inline src/compas/rpc/services/default.py]: delete

File: src/compas/rpc/proxy.py [hunk 2]
```diff
@@ -56,6 +56,12 @@ class Proxy(object):
     capture_output : bool, optional
         If True, capture the stdout/stderr output of the remote process.
         In general, `capture_output` should be True when using a `pythonw` as executable (default).
+    path : str, optional
+        Path to the folder containing the module to be proxied.
+        This is useful for cases where the module to be proxied is not on the PYTHONPATH.
+    cwd : str, optional
+        Current working directory for the process that will be started to run the server.
+        This is useful for cases where a custom service is used and the service is not on the PYTHONPATH.
```
Comment (gonzalocasas) [inline src/compas/rpc/proxy.py]: Isn't `cwd` too cryptic for most people? Shouldn't we just err on the side of verbosity and call it `working_directory`?

File: src/compas/rpc/services/special.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+"""
+Minimal example of a "special" service.
```
Comment (gonzalocasas) [inline src/compas/rpc/services/special.py]: I'm not very convinced about the naming, "special" but why? What's special about it? :) Do you mean "Custom" or "UserDefined"?

**Review by gonzalocasas — APPROVED** [review]
lgtm + feedback

File: src/compas/rpc/services/special.py (hunk 3 above)
Comment (tomvanmele) [reply src/compas/rpc/services/special.py]: not it is just an example. perhaps it should be removed and the example added to the docs

Comment (gonzalocasas) [reply src/compas/rpc/services/special.py]: 👍

**Pull request merged**